### DISCUSSION TITLE
I am using take_foucus and it has no effect

### DISCUSSION THREAD

**Discussion by Lydanne** [fltk-rs/fltk-rs#1257] [Q&A] [answered]
I want the input to enter the text directly, instead of typing after clicking, I am using take_foucus and it has no effect.

```rust
  self.input.set_size(0, 0);
  self.input.set_value("");
  self.input.take_focus().unwrap();
```
To run the code please run packages/drawpanel
link:https://github.com/drawpanel/drawpanel/blob/debug/fltk-input/packages/drawpanel-bind-fltk/src/lib.rs#L186

**Comment by Lydanne** [comment, marked as answer]
my question

**Reply by MoAlyousef** [reply]
You can try the app::set_focus function (which sets which widget has focus on app startup):
https://docs.rs/fltk/latest/fltk/app/fn.set_focus.html

take_focus is mostly used within event handling.

**Reply by Lydanne** [reply]
This is my problem. I didn't control the timing.
